hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/NoSuchMessageTask.java

```diff
@@ -44,8 +44,10 @@ protected ClientMessage encodeResponse(Object response) {
 
     @Override
     protected void processMessage() {
-        logger.warning("Unrecognized client message received with type: 0x"
-                + Integer.toHexString(parameters.getMessageType()));
+        String message = "Unrecognized client message received with type: 0x"
+                + Integer.toHexString(parameters.getMessageType());
+        logger.warning(message);
+        throw new UnsupportedOperationException(message);
```

Comment:
Yes, caller is going to get `UnsupportedOperationException` exception. 
I have just updated the test to use same method to peel exception when used over a proxy as our users will do.